### PR TITLE
Clean env

### PR DESCRIPTION
**Pull request by bartekGardas**
Small improvements that may speed SpinGlassEngine

### CODE REVIEW

File: src/identities.jl [hunk 1]
```diff
@@ -12,33 +12,29 @@ IdentityMPO(::Type{S}, ::Type{T}) where {S <: Number, T <: AbstractArray} = Iden
...
-    for i ∈ eachindex(ψ)
-        B = O[i]
-        @reduce A[x, σ, y] |= sum(η) B[x, σ, y, η]
-        ψ[i] = A
-    end
-    ψ
-end
+
+LinearAlgebra.dot(O::AbstractMPO, ::IdentityMPS) = 
+MPS([dropdims(sum(A, dims=4), dims=4) for A ∈ O])
```
Comment (lpawela) [inline src/identities.jl:36]: Please add explicit element type for `MPS`

File: src/contractions.jl [hunk 2]
```diff
@@ -120,34 +115,18 @@ end
...
+    ])
 end
 
 
 function LinearAlgebra.dot(O1::AbstractMPO, O2::AbstractMPO)
-    S = promote_type(eltype(O1), eltype(O2))
-    T = typeof(O1)
-    O = T.name.wrapper(S, length(O1))
-    for (i, (A, B)) ∈ enumerate(zip(O1, O2))
+    MPO([
```
Comment (lpawela) [inline src/contractions.jl]: This will fail with `CuMPO`, please restore previous approach.

File: src/contractions.jl [hunk 3]
```diff
@@ -120,34 +115,18 @@ end
 
 
 function LinearAlgebra.dot(O::AbstractMPO, ψ::AbstractMPS)
-    S = promote_type(eltype(ψ), eltype(O))
-    T = typeof(ψ)
-    ϕ = T.name.wrapper(S, length(ψ))
-    for (i, (A, B)) ∈ enumerate(zip(O, ψ))
+    MPS([
```
Comment (lpawela) [inline src/contractions.jl]: This will fail with `CuMPS`, please restore previous approach.

**Review by lpawela — CHANGES_REQUESTED** [review]

File: src/identities.jl (hunk 1 above)
Comment (bartekGardas) [reply src/identities.jl:36]: This should promote a correct type.

File: src/contractions.jl (hunk 2 above)
Comment (bartekGardas) [reply src/contractions.jl]: But this should be faster - less copy operations if I'm not mistaken.

File: src/contractions.jl (hunk 3 above)
Comment (bartekGardas) [reply src/contractions.jl]: Same as above.

**Review by lpawela — APPROVED** [review]

**Pull request merged**